staging/src/k8s.io/apiserver/pkg/server/options/encryptionconfig/config.go

```diff
@@ -258,3 +259,18 @@ func GetSecretboxPrefixTransformer(config *SecretboxConfig) (value.PrefixTransfo
 	}
 	return result, nil
 }
+
+func readUptoLength(f io.Reader, len int64) ([]byte, error) {
+	contents, err := ioutil.ReadAll(io.LimitReader(f, len))
+	if err != nil {
+		return nil, err
+	}
+	_, err = io.ReadAtLeast(f, []byte{0}, 1)
```

Comment:
IMO we need further discussion first on whether we'll be reading upto a maximum and be strict about it.